superset/connectors/base/models.py

```diff
@@ -308,8 +307,8 @@ def data_for_slices(self, slices: List[Slice]) -> Dict[str, Any]:
 
             column_names.update(
                 column
-                for column in utils.get_iterable(form_data.get(param) or [])
-                for param in COLUMN_FORM_DATA_PARAMS
+                for column_param in COLUMN_FORM_DATA_PARAMS
+                for column in utils.get_iterable(form_data.get(column_param) or [])
```

Comment:
The order of the for loop is incorrect. Normally it will throw an undefined variable error but since `param` was used in other loops above, the code will still run. Just another example of why too many local variables is not a good thing.